src/Models/Services/LocalStorageManager/LocalStorageManager.ts

```diff
@@ -0,0 +1,331 @@
+    IAzureResource,
+    IAzureStorageAccount,
+    IAzureStorageBlobContainer
+} from '../../Constants/Interfaces';
+import { getNameOfResource, getResourceUrl } from '../Utils';
+import {
+    EnvironmentConfigurationInLocalStorage,
+    EnvironmentConfigurationItem,
+    EnvironmentConfigurationLocalStorageKey
+} from './LocalStorageManager.types';
+
+/** This is used for converting a adt, storage account or storage container resource
+ * to a environment configuration item to store in localstorage
+ * @param id id of the resource
+ * @param name name of the resource
+ * @param url url of the resource
+ */
+export const getEnvironmentConfigurationItemFromResource = (
+    resource: IAzureResource | string,
+    resourceType: AzureResourceTypes,
+    parentResource?: IAzureResource | string // to construct the url for a container we need its parent storage account's url
+): EnvironmentConfigurationItem | null => {
+    if (!resource) return null;
+    return {
+        id: typeof resource === 'string' ? null : resource.id,
+        name: getNameOfResource(resource, resourceType),
+        url: getResourceUrl(resource, resourceType, parentResource)
+    };
+};
+
+/** This is used for getting the main environment configuration local storage item which stores
+ * adt, storage account and storage container related information in it
+ * @returns te environment configuration local storage item or null if not exists
+ */
+export const getEnvironmentConfigurationFromLocalStorage = (): EnvironmentConfigurationInLocalStorage | null => {
+    try {
+        const environmentConfigurationInLocalStorage = localStorage.getItem(
+            EnvironmentConfigurationLocalStorageKey
+        );
+        return environmentConfigurationInLocalStorage
+            ? (JSON.parse(
+                  environmentConfigurationInLocalStorage
+              ) as EnvironmentConfigurationInLocalStorage)
+            : null;
+    } catch (error) {
+        console.error(error.message);
+        return null;
+    }
+};
+
+/** To get the list of ADT instances from local storage to be used as options in EnvironmentPicker
+ * @return list of ADT instances as environment configuration item from local storage
+ */
+export const getAdtInstancesFromLocalStorage = (): Array<EnvironmentConfigurationItem> | null => {
+    try {
+        const environmentConfigurationInLocalStorage: EnvironmentConfigurationInLocalStorage = JSON.parse(
+            localStorage.getItem(EnvironmentConfigurationLocalStorageKey)
+        );
+        return environmentConfigurationInLocalStorage?.adt?.adtInstances;
+    } catch (error) {
+        console.error(error.message);
+        return null;
+    }
+};
+
+/** To get the list of ADT instances from local storage to be used as options in EnvironmentPicker
+ * @return list of ADT instances as environment configuration item from local storage
+ */
+export const getStorageAccountsFromLocalStorage = (): Array<EnvironmentConfigurationItem> => {
```

Comment:
LocalStorageManager is not a class by itself, since methods there individually exposed, I would rather to see the "..localStorage" in the name of the methods when I consume it in my components. There might be other methods exported with names "getStorageAccounts" which maybe conflicted. So, what do you think? I prefer to keep them.

Reply:
ok. not a big deal either way. Fine to leave them.